azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1340,14 +1190,12 @@ func flattenKubernetesClusterAgentPoolProfiles(profiles *[]containerservice.Mana
 	return agentPoolProfiles
 }
 
-func expandKubernetesClusterLinuxProfile(d *schema.ResourceData) *containerservice.LinuxProfile {
-	profiles := d.Get("linux_profile").([]interface{})
-
-	if len(profiles) == 0 {
+func expandKubernetesClusterLinuxProfile(input []interface{}) *containerservice.LinuxProfile {
+	if len(input) == 0 {
 		return nil
 	}
 
-	config := profiles[0].(map[string]interface{})
+	config := input[0].(map[string]interface{})
```

Comment:
We should check that this isn't nil

Reply:
this should be fine - since it's valid for this to have no elements (so this should either be a concrete value or empty)